src/panels/lovelace/components/hui-image.ts

```diff
@@ -156,46 +190,88 @@ export class HuiImage extends LitElement {
                 @load=${this._onImageLoad}
                 style=${styleMap({
                   filter,
-                  display: this._loadError ? "none" : "block",
+                  display:
+                    this._loadState === LoadState.Loaded ? "block" : "none",
                 })}
               />
             `}
-        ${this._loadError
+        ${this._loadState === LoadState.Error
           ? html`<div
               id="brokenImage"
               style=${styleMap({
                 height: `${this._lastImageHeight || "100"}px`,
               })}
             ></div>`
+          : imageSrc === undefined || this._loadState === LoadState.Loading
+          ? html`<div
+              style=${styleMap({
+                height: `${this._lastImageHeight || "100"}px`,
+              })}
+            >
+              <ha-circular-progress
+                class="render-spinner"
+                active
+                size="small"
+              ></ha-circular-progress>
+            </div>`
           : ""}
       </div>
     `;
   }
 
+  protected _shouldStartCameraUpdates(oldHass?: HomeAssistant): boolean {
+    return !!(
+      (!oldHass || oldHass.connected !== this.hass!.connected) &&
+      this.hass!.connected &&
+      this.cameraView !== "live"
+    );
+  }
+
   protected updated(changedProps: PropertyValues): void {
```

Comment:
You can move everything in `updated` to `willUpdate`

Reply:
moved into willUpdate